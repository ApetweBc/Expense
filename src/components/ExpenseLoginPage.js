import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';



const ExpenseLoginPage = ({ startLogin }) =>  (

    <div>
       <button onClick={startLogin}> Login </button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
       startLogin:  () => dispatch(startLogin())
});
export default connect (undefined, mapDispatchToProps)(ExpenseLoginPage);
