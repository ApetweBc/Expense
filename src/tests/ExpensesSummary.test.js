
import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../src/components/ExpensesSummary';

test('Should correctly render Expenses Summary with 1 Expense',()=>{
   const wrapper = shallow (<ExpensesSummary expenseCount={1} expensesTotal={235} />);
   expect(wrapper).toMatchSnapshot();
});


test('Should correctly render Expenses Summary with multipleExpense Expense',()=>{
    const wrapper = shallow (<ExpensesSummary expenseCount={23} expensesTotal={2354323523232} />);
    expect(wrapper).toMatchSnapshot();
 });
