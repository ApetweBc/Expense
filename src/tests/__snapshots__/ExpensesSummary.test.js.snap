// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Expenses Summary with 1 Expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`Should correctly render Expenses Summary with multipleExpense Expense 1`] = `
<div>
  <h1>
     Viewing 
    23
     
    expenses
     totalling 
    $23,543,235,232.32
  </h1>
</div>
`;
